lib/jsonapi-serializers/serializer.rb

```diff
@@ -359,10 +359,15 @@ def self.serialize_primary(object, options = {})
 
       serializer = serializer_class.new(object, options)
       data = {
-        'id' => serializer.id.to_s,
         'type' => serializer.type.to_s,
       }
 
+      # "The id member is not required when the resource object originates at the client
+      #  and represents a new resource to be created on the server."
+      # http://jsonapi.org/format/#document-resource-objects
+      # We'll assume that if the id is blank, it means the resource is to be created.
+      data['id'] = serializer.id.to_s if !serializer.id.empty?
```

Comment:
I still have to do `!...empty?` here, because the default serializer `id` method coerces to string.  The second `.to_s` here was originally redundant, so I've just removed that.

There's still a problem here though, if the user overrides the the `id` method to sometimes return nil instead of always returning a string, we'll get a NoMethodError...  maybe leaving the `.to_s` there is better for safety?  Thoughts?